infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/StreamsLogEntryReader.java

```diff
@@ -146,6 +158,14 @@ private boolean isValidTransactionEntry(@NonNull OpaqueEntry entry) {
             return false;
         }
 
+        // In DYNAMIC mode, chances are that some streams to replicate were not registered when LogReplicationConfig
+        // was initiated. If txEntryStreamIds contains additional streams, config needs to sync with registry table
+        // to avoid data loss.
+        if (config.getStreamsDiscoveryMode().equals(StreamsDiscoveryMode.DYNAMIC) &&
+                !streamUUIDs.containsAll(txEntryStreamIds)) {
+            config.syncWithRegistry();
```

Comment:
this will read the whole registry table again, populate the streams to replicate, tags map etc.  We only need to update with a few newly discovered streams.  I suggest having a new method which just updates the the list with the new streams by checking the registry table.

Reply:
discussed offline and decided to use this method as the operation is inexpensive.